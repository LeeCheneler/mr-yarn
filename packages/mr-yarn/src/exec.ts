import { exec as doExec, ExecOptions } from 'child_process'
import { promisify } from 'util'

/**
 * Exec a command
 * @param command Command to execute
 * @param options
 */
export const exec = async (command: string, options: ExecOptions) => {
  const result = await promisify(doExec)(command, options)

  if (result.stderr) {
    throw new Error(result.stderr)
  }

  return result
}
