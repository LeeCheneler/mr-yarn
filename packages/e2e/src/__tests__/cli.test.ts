import { resolve } from 'path'
import { exec } from '../utils/exec'

/**
 * Fixtures directory containing variants of workspace setup to assert against
 */
const fixturesDir = resolve(process.cwd(), 'src/__fixtures__')
const cwd = resolve(fixturesDir, 'valid')

it('Should print version', async () => {
  const { stdout } = await exec('mr --version', { cwd })

  expect(stdout).toMatchSnapshot()
})

it('Should print help', async () => {
  const { stdout } = await exec('mr --help', { cwd })

  expect(stdout).toMatchSnapshot()
})
