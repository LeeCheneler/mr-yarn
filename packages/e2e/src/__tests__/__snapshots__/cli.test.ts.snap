// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should print help 1`] = `
"mr [command]

Commands:
  mr add [packages...]          Add packages to workspaces
  mr run <script> [options...]  Run NPM script in workspaces
  mr remove [packages...]       Remove packages from workspaces

Options:
  -w, --workspaces  Filter by workspace names. Defaults to all when option not
                    present.
  --help            Show help                                          [boolean]
  --version         Show version number                                [boolean]
"
`;

exports[`Should print version 1`] = `
"1.0.0-alpha.5
"
`;
